tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -38,47 +42,78 @@ typedef struct _connectionHandle
   char data[USER_DATA_SIZE];
 } _connHandle_t;
 
-/* Models the third party HTTP Parser. */
+/****************************************************************
+ * HTTP parser stubs
+ ****************************************************************/
+
+/* Model the third party HTTP Parser. */
 size_t http_parser_execute (http_parser *parser,
                             const http_parser_settings *settings,
                             const char *data,
                             size_t len) {
-  size_t ret;
+  __CPROVER_assert(parser, "http_parser_execute parser nonnull");
+  __CPROVER_assert(settings, "http_parser_execute settings nonnull");
+  __CPROVER_assert(data, "http_parser_execute data nonnull");
+
   _httpsResponse_t *_httpsResponse = (_httpsResponse_t *)(parser->data);
   /* nondet_bool is required to get coverage. */
   _httpsResponse->foundHeaderField = nondet_bool();
   _httpsResponse->parserState = PARSER_STATE_BODY_COMPLETE;
+
+  size_t ret;
```

Comment:
In the interest of non-CBMC people who read this: suggest
```c
size_t ret = nondet_size_t();
return ret;
```
(not sure if we have a `nondet_size_t()` implementation, but please add one if not)

Reply:
Or just ```return nondet_size_t();``` . In fact, I could define nondet_MQTT_bool() and do the same thing in all of the stubs in global_state_HTTP.

Reply:
That would make it easier to read. But looking at the rest of the code it is easy to see that these primitives by themselves become non-determinate. So I am ok with leaving as is.